src/transaction/log_recovery_analysis.cpp

```diff
@@ -2335,30 +2373,53 @@ log_recovery_build_mvcc_table_from_trantable (THREAD_ENTRY *thread_p)
 		  largest_mvccid = tdes->mvccinfo.id;
 		}
 	      present_mvccids.insert (tdes->mvccinfo.id);
+
+	      // TODO: for later, mvcc sub ids
+	      assert (tdes->mvccinfo.sub_ids.empty ());
 	    }
 	}
     }
-  log_Gl.hdr.mvcc_next_id = smallest_mvccid;
-  log_Gl.mvcc_table.reset_start_mvccid ();
 
-  if (!present_mvccids.empty ())
+  if (smallest_mvccid == max_possible_mvccid && largest_mvccid == min_possible_mvccid)
```

Comment:
```suggestion
  if (smallest_mvccid == max_possible_mvccid || largest_mvccid == min_possible_mvccid)
```

Reply:
In practice, if either of them is initialized, the other is initialized as well. So the conditions are - somewhat - equivalent. However, I added an assert along these lines:
```
  assert ((max_possible_mvccid == smallest_mvccid && min_possible_mvccid == largest_mvccid)
	  || (max_possible_mvccid != smallest_mvccid && min_possible_mvccid != largest_mvccid));
```